src/local/cost_matrix.rs

```diff
@@ -61,6 +67,37 @@ impl LocalCostMatrix {
         &self.bits
     }
 
+    pub fn iter(&self) -> impl Iterator<Item = ((u8, u8), &u8)> {
+        self.bits.iter().enumerate().map(|(idx, val)| { (idx_as_pos(idx), val) })
+    }
+
+    pub fn iter_mut(&mut self) -> impl Iterator<Item = ((u8, u8), &mut u8)> {
+        self.bits.iter_mut().enumerate().map(|(idx, val)| { (idx_as_pos(idx), val) })
+    }
+
+    // Takes all non-zero entries in `src`, and inserts them into `self`.
+    //
+    // If an entry for that position exists already, overwrites it with the new
+    // value.
+    pub fn merge_from_dense(&mut self, src: &LocalCostMatrix) {
```

Comment:
It might make more sense to create a 'merge into' trait that the receiving matrix can implement? Having to match the type with the right function is a bit less ergonomic.